File: PE1 - Module 4/Fibonacci.java
public final class Fibonacci {

    public static void main(String[] args) {
        var number = Integer.parseInt(args[0]);
        System.out.println(number + " -> " + fibonacci(number));
    }

    private static final int fibonacci(int n) {
    
        if (n < 0) {
            return 0;
        }

        if (n < 3) {
            return 1;
        } 
       
        return fibonacci(n - 1) + fibonacci(n - 2);
    }
}

File: PE1 - Module 4/fibonacci.py
def fibonacci(n):
    if n < 1:
        return None

    if n < 3:
        return 1

    result = 0
    item1 = 1
    item2 = 1
    
    for i in range(3, n + 1):
        result = item1 + item2
        item1, item2 = item2, result

    return result

# Testing
#for i in range(1, 10):
#    print(i, fibonacci(i))

number = int(input("Enter the number to calculate its Fibonacci: "))
print("Fibonacci of", number, "is", fibonacci(number))
